### DISCUSSION TITLE
I want to fill the black area in the figure below with an image. How can I do it? If not, can I fill it with color

### DISCUSSION THREAD

**Discussion by jueqing1015** [dlemstra/Magick.NET#1017] [Help]
![Sample-0005](https://user-images.githubusercontent.com/65677359/131935495-f0d9951e-0111-4e95-8299-62db3475f3ba.jpg)
I want to fill the black area around the figure above with an image. If it cannot be filled with an image, can I fill it with white,

**Comment by dlemstra** [comment]
This might now work for every image but you could do something like this:

```C#
using (var image = new MagickImage("input.jpg"))
{
    // Add a border on all sides of the image to make it possible to fill the image.
    image.Extent(image.Width + 2, image.Height + 2, Gravity.Center, MagickColors.Black);

    // Fill the black border pixels with a transparent color.
    image.ColorFuzz = new Percentage(10);
    image.FloodFill(MagickColors.Transparent, 0, 0);

    // Trim the image to remove the border that was added.
    image.Trim();

    // Load background image, use your own background instead of the sample below
    using (var background = new MagickImage("plasma:", image.Width, image.Height))
    {
        // Draw background under the image.
        image.Composite(background, CompositeOperator.DstAtop);
    }

    image.Write("output.jpg");
}
```

**Reply by jueqing1015** [reply]
@dlemstra 
[新建文件夹 (4).zip](https://github.com/dlemstra/Magick.NET/files/7229170/4.zip)

Thank you very much for your answer. I found a problem through the test. If the black area of its border is pure black, it is no problem, but if it is not so black, it is not very clean. I don't know whether it is necessary to set parameters or other methods

I want to process image.jpg to image2.jpg, and image1 is the current processed effect

**Reply by dlemstra** [reply]
The `image.ColorFuzz` property can be used for this. You will probably need to increase it a bit.

**Reply by jueqing1015** [reply]
@dlemstra Yes, but it will be very slow. Can I use a mask

**Reply by dlemstra** [reply]
When you make the background a mask you could draw that on top of the image.

**Reply by jueqing1015** [reply]
@dlemstra 
How should I operate it, thank you

The demand for this function is too urgent. If I set colorfuzz, it can be realized, but the efficiency will be greatly reduced. Please give me some advice,

**Reply by dlemstra** [reply]
Do you have the mask already? I am not sure what you are trying to ask me.

**Reply by jueqing1015** [reply]
Not at present. I just want to remove the black edge by mask

**Reply by dlemstra** [reply]
If you have an image and a mask I can explain you what to do. But you will need to manually create a mask for every type of image.

**Reply by jueqing1015** [reply]
https://github.com/dlemstra/Magick.NET/files/7229170/4.zip
Could you please explain it with the image inside

**Reply by dlemstra** [reply]
You don't have a mask in those files. There is no method to just create the mask, you will need to do that manually.

**Reply by jueqing1015** [reply]
https://github.com/DanBloomberg/leptonica/issues/590
I use another control to achieve the desired effect, but now I want to go to magic.net. Now I have done some of the steps. Please help me see. I'm one step away now


MagickImage image = new MagickImage("c:\\0025.jpg");
            image.Threshold(new Percentage(50));
            MagickImage mask = new MagickImage(image.Clone());
            MagickImage newmask = new MagickImage(mask.Clone());
            mask.Threshold(new Percentage(70));
            mask.BorderColor = MagickColors.Black;
            mask.Border(1);

            mask.ColorFuzz = new Percentage(10);
            mask.FloodFill(MagickColors.White, 0, 0);
            newmask.Composite(mask, CompositeOperator.DstAtop);
            image.Composite(newmask, CompositeOperator.CopyAlpha);
            MagickImage images = new MagickImage(MagickColors.White, image.Width, image.Height);
            image.Composite(mask, CompositeOperator.DstAtop);
